src/Http/Http/src/Timeouts/RequestTimeoutsMiddleware.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http.Timeouts;
+
+internal sealed class RequestTimeoutsMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly ICancellationTokenLinker _cancellationTokenProvider;
+    private readonly ILogger<RequestTimeoutsMiddleware> _logger;
+    private readonly IOptionsMonitor<RequestTimeoutOptions> _options;
+
+    public RequestTimeoutsMiddleware(
+        RequestDelegate next,
+        ICancellationTokenLinker cancellationTokenProvider,
+        ILogger<RequestTimeoutsMiddleware> logger,
+        IOptionsMonitor<RequestTimeoutOptions> options)
+    {
+        _next = next;
+        _cancellationTokenProvider = cancellationTokenProvider;
+        _logger = logger;
+        _options = options;
+    }
+
+    public Task Invoke(HttpContext context)
+    {
+        var endpoint = context.GetEndpoint();
+        var timeoutMetadata = endpoint?.Metadata.GetMetadata<RequestTimeoutAttribute>();
+        var policyMetadata = endpoint?.Metadata.GetMetadata<RequestTimeoutPolicy>();
+
+        var options = _options.CurrentValue;
+
+        if (timeoutMetadata is null && policyMetadata?.Timeout is null && options.DefaultPolicy?.Timeout is null)
+        {
+            return _next(context);
+        }
+
+        var disableMetadata = endpoint?.Metadata.GetMetadata<DisableRequestTimeoutAttribute>();
```

Comment:
should we also disable timeouts if the debugger is attached? That's something we do in kestrel to make debugging easier.
https://github.com/dotnet/AspNetCore.Docs/pull/28212/files

Reply:
I'm so torn here, but we do it everywhere else so probably.

Reply:
Does this mean that we skip the middleware if debugger is attached? The only problem that I see is that users can't run `WriteTimeoutResponse` while developing and this might be confusing.

Reply:
Yes